reactive/calico.py

```diff
@@ -379,31 +387,35 @@ def read_file_to_base64(path):
 
 
 def calicoctl(*args):
+    directories = [
+        '/var/run/calico',
+        '/var/lib/calico',
+        '/run/containerd/plugins',
+        '/var/log/calico'
+    ]
+
+    for d in directories:
+        os.makedirs(d, exist_ok=True)
+
     etcd = endpoint_from_flag('etcd.available')
     registry = hookenv.config('registry') or 'quay.io'
-    image = registry + '/tigera/calicoctl:v2.3.0'
-    cmd = [
-        'docker', 'run',
-        '-v', CALICOCTL_PATH + ':' + CALICOCTL_PATH,
-        '-v', '/tmp:/tmp',
-        '-e', 'ETCD_ENDPOINTS=' + etcd.get_connection_string(),
-        '-e', 'ETCD_KEY_FILE=' + ETCD_KEY_PATH,
-        '-e', 'ETCD_CERT_FILE=' + ETCD_CERT_PATH,
-        '-e', 'ETCD_CA_CERT_FILE=' + ETCD_CA_PATH,
-        image
-    ]
-    cmd += list(args)
-    try:
-        return check_output(cmd)
-    except CalledProcessError as e:
-        log(e.output)
-        raise
-
-
-def arch():
-    '''Return the package architecture as a string.'''
-    # Get the package architecture for this system.
-    architecture = check_output(['dpkg', '--print-architecture']).rstrip()
-    # Convert the binary result into a string.
-    architecture = architecture.decode('utf-8')
-    return architecture
+    image = registry + CALICO_IMAGE
+
+    CTL.run(
+        privileged=True,
+        net_host=True,
```

Comment:
Why `privileged=True` and `net_host=True`?

Reply:
`net_host` seems required. otherwise:

```
Failed to create Calico API client: dial tcp 172.31.47.29:2379: connect: network is unreachable
```